src/mpi/romio/adio/ad_lustre/ad_lustre_wrcoll.c

```diff
@@ -120,6 +120,7 @@ void ADIOI_LUSTRE_WriteStridedColl(ADIO_File fd, const void *buf, int count,
     ADIO_Offset *lustre_offsets0, *lustre_offsets, *count_sizes = NULL;
 
     MPI_Comm_size(fd->comm, &nprocs);
+    fd->hints->cb_nodes = nprocs;
```

Comment:
@liy106 here's the problem!    Welcome to ROMIO where nothing is adequately documented.  There is a fd->hints->ranklist[] array allocated to be cb_nodes big.  By increasing cb_nodes here, you will overrrun that array elsewhere.